src/sgp/SGPFile.cc

```diff
@@ -0,0 +1,297 @@
+#include "SGPFile.h"
+#include "RustInterface.h"
+#include "Exceptions.h"
+#include "Logger.h"
+
+#include <string_theory/string>
+#include <string_theory/format>
+
+#define SDL_RWOPS_SGP 222
+
+void DeleteSGPFile(SGPFile *file)
```

Comment:
Maybe not this time, but next time we can refactor away the `AutoSGPFile` class and this function.

Reply:
Yes, I had started with that, but that would have bloated the PR even more.